arbos/tx_processor.go

```diff
@@ -628,8 +627,15 @@ func (p *TxProcessor) L1BlockHash(blockCtx vm.BlockContext, l1BlockNumber uint64
 	return hash, nil
 }
 
+func (p *TxProcessor) DropTip() bool {
+	return p.state.ArbOSVersion() < 8 || p.delayedInbox
+}
+
 func (p *TxProcessor) GetPaidGasPrice() *big.Int {
-	gasPrice := p.evm.Context.BaseFee
+	gasPrice := p.evm.GasPrice
+	if p.state.ArbOSVersion() < 8 {
+		gasPrice = p.evm.Context.BaseFee
+	}
 	if p.msg.RunMode() != types.MessageCommitMode && p.msg.GasFeeCap().Sign() == 0 {
 		gasPrice.SetInt64(0) // gasprice zero behavior
```

Comment:
This check should only be necessary for the old arbos versions where we were using the base fee

Reply:
Nested